lib/sidekiq/superworker/subjob_processor.rb

```diff
@@ -110,7 +110,9 @@ def descendants_are_complete(subjob)
             if siblings_descendants_are_complete
               Superworker.debug "#{subjob.to_info}: Parent (#{parent.to_info}) is complete"
               descendants_are_complete(parent)
-              parent.update_attribute(:status, 'complete') if is_child_of_parallel
+              if is_child_of_parallel && !Superworker.options[:delete_subjobs_after_superjob_completes]
```

Comment:
If we have delete subjobs enabled, we will delete the parent subjob but then write the job again to redis saying its complete. This should catch that edgecase.